src/clr-angular/forms/datepicker/date-container.ts

```diff
@@ -122,12 +137,30 @@ export class ClrDateContainer implements DynamicWrapper, OnDestroy, AfterViewIni
     );
   }
 
+  /**
+   * Clear datapicker value
+   */
+  public clearDateValue() {
+    this.dateIOService.clearInputDate();
+  }
+
   ngOnInit() {
     this.subscriptions.push(
       this.ifErrorService.statusChanges.subscribe(invalid => {
         this.invalid = invalid;
       })
     );
+
+    /** handle when to show are hide the clear button */
+    this.subscriptions.push(
```

Comment:
Why not use an @ContentChild for the clrDate element and subscribe to value changes there? It might eliminate the need for the service changes.